src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/UsageBasedMetadataManager.cs

```diff
@@ -563,6 +563,8 @@ protected override void GetDependenciesDueToMethodCodePresenceInternal(ref Depen
                 {
                     AddDataflowDependency(ref dependencies, factory, methodIL, "Method has annotated parameters");
                 }
+
+                StaticConstructorAnalysisNode.GetDependencies(ref dependencies, factory, method);
```

Comment:
Why don't we do the check here? (Why is the new node needed?)

Reply:
Can we guarantee that this is only called once per method? The new node is to avoid duplicate warnings.

Reply:
I think so - this callback is called when a method body is generated. We only generate them once.

Reply:
OK- I'll look around and make the change then.

Reply:
It can happen that it will be called more than once (I think) for GVMs, the `RuntimeMethodHandleNode` will call this on the cannon abstract GVM method - and that can probably happen more than once if it finds more than one instantiation.

But since the new code only cares about static cctor, that has no effect (cctor is never going to be a GVM).

I simplified this change by removing the new node.